### ISSUE TITLE
Show hardware metrics per job (e.g. CPU, memory, and GPU utilization)

### ISSUE THREAD

**Issue by kengz** [dstackai/dstack#82]
This will help with debugging and understanding resources usage, also general diagnostics:
- cpu available vs utilized
- gpu available vs utilized
- memory available vs utilized
- total runtime of container

**Comment by Adesoji1** [comment]
this is what i did and it worked, using the codes below

```
import psutil
import subprocess
import time

# CPU metrics
cpu_percent = psutil.cpu_percent()
cpu_count = psutil.cpu_count()
cpu_stats = psutil.cpu_stats()

# Memory metrics
memory = psutil.virtual_memory()
memory_percent = memory.percent
memory_used = memory.used
memory_available = memory.available

# GPU metrics
command = 'nvidia-smi --query-gpu=utilization.gpu --format=csv,noheader,nounits'
result = subprocess.run(command, capture_output=True, text=True, shell=True)
gpu_utilization = result.stdout.strip()

# Start time
start_time = time.time()

# Your code or process here

# End time
end_time = time.time()
runtime = end_time - start_time

# Print metrics
print("CPU Utilization:", cpu_percent)
print("CPU Count:", cpu_count)
print("CPU Stats:", cpu_stats)
print("Memory Utilization:", memory_percent)
print("Memory Used:", memory_used)
print("Memory Available:", memory_available)
print("GPU Utilization:", gpu_utilization)
print("Total Runtime:", runtime, "seconds")
```

Output Below

```

CPU Utilization: 4.5
CPU Count: 12
CPU Stats: scpustats(ctx_switches=38669011, interrupts=15883059, soft_interrupts=14444502, syscalls=0)
Memory Utilization: 65.5
Memory Used: 8727457792
Memory Available: 5685366784
GPU Utilization: 0
Total Runtime: 2.002716064453125e-05 seconds
```

When you run this code, it will display the CPU utilization, CPU count, CPU stats, memory utilization, memory used, memory available, GPU utilization, and the total runtime of the container.

Note that the GPU metrics part assumes you have the NVIDIA System Management Interface (nvidia-smi) installed on your system and available in your system's PATH. If you don't have GPUs or the required dependencies, you can still run the code, but the GPU utilization part will not provide any meaningful results.

**Comment by r4victor** [comment]
`dstack run` now shows available CPU, memory, and GPU (#459)
